export_workspace/eager_decomp.py

```diff
@@ -0,0 +1,34 @@
+import torch
+from torch.utils._python_dispatch import TorchDispatchMode
+from torch._decomp import core_aten_decompositions
+
+class TestModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.bn = torch.nn.BatchNorm2d(3)
+
+    def forward(self, x):
+        return self.bn(x)
+
+x = torch.rand(2, 3, 244, 244)
+model = TestModule()
+
+class TorchEagerDecomposeMode(TorchDispatchMode):
+    def __init__(self, decomposition_table):
+        super().__init__()
+        self.decomposition_table = decomposition_table
+
+    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
+        if func in self.decomposition_table:
+            print(f"Decomposing {func}")
+            with self:
+                return self.decomposition_table[func](*args, **kwargs)
```

Comment:
Do you want to add
```
        with self:
            result = func.decompose(*args, **kwargs)
            if result is not NotImplemented:
                return result
```
, which triggers scalar to tensor type promotion?

Reply:
If there are multiple ways of doing decomposition, it'd be helpful to discuss them in one place (like here).